frontend/ui/widget/keyvaluepage.lua

```diff
@@ -543,6 +544,90 @@ function KeyValuePage:_populateItems()
     self.return_button:resetLayout()
     self.main_content:clear()
     local idx_offset = (self.show_page - 1) * self.items_per_page
+
+    -- for flexible middle ratio calculation
+    local frame_padding = Size.padding.default
+    local frame_internal_width = self.item_width - frame_padding * 2
+    local middle_padding = Size.padding.default -- min enforced padding between key and value
+    local available_width = frame_internal_width - middle_padding
+    -- Default widths (and position of value widget) if each text fits in 1/2 screen width
+    local key_w = math.floor(frame_internal_width / 2 - middle_padding)
+    local value_w = math.floor(frame_internal_width / 2)
+
+    local key_widget = TextWidget:new{
+        text = " ",
+        max_width = available_width,
+        face = Font:getFace("smallinfofontbold", self.items_font_size),
+    }
+    local value_widget = TextWidget:new{
+        text = " ",
+        max_width = available_width,
+        face = Font:getFace("smallinfofont", self.items_font_size),
+        lang = self.values_lang,
+    }
+    local key_widths = {}
+    local value_widths = {}
+    local tvalue
+    for idx=1, self.items_per_page do
+        local kv_pairs_idx = idx_offset + idx
+        local entry = self.kv_pairs[kv_pairs_idx]
+        if entry == nil then break end
+        if type(entry) == "table" then
+            tvalue = tostring(entry[2])
+            tvalue = tvalue:gsub("[\n\t]", "|")
+
+            key_widget:setText(entry[1])
+            value_widget:setText(tvalue)
+
```

Comment:
(No blank lines neeeded.)
Also don't forget to `key_widget:free()` when you're done with these 2 TextWidgets.

Reply:
I did forgot about it. Got too used to auto-released objects :)

Reply:
I think they would be garbage collected, later, and they would be then cleaned. But when we can, best to use free() to release some resources (the xtext.cpp objects here) without having to wait until the next gc.